services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/nodeports.py

```diff
@@ -170,6 +170,76 @@ async def _get_data_from_port(port: Port) -> Tuple[Port, ItemConcreteValue]:
     return (port, ret)
 
 
+async def _download_files(
+    target_path: Path, download_tasks: Deque[Coroutine[Any, int, Any]]
+) -> Tuple[dict[str, Any], ByteSize]:
```

Comment:
1. Could you create a TypeDict at least, to know what is returned?
2. check if you can use deck instead of Deque
3. you can use tuple instead of Tuple
4. if you bother on butting a deque here, why then not just pass a Sequence type? I mean the best and most secure/fast way is to pass a tuple because then you also pass the fact that the fct will not modify the argument, plus tuple is faster than deque... also I doubt a bit that going from list to deque brings so much of a difference... especially when downloading GBs...

Reply:
1. I've found out we already have something similar defined `OutputsDict`. Will be using that.
2. Sorry I don't understand the suggestion.
3. 👍 
4. There is no life changing improvement. The deque is just a better suited for appending data than a list. The change from list to deuque brings no real life benefit. I do not agree on the use of tuples, since they are immutable ad when constructing the sequence you need to append data to it.

Reply:
2. `deque` is usable for typing